docs: update the section about zsh completions (#5825)
ddev/ddev: docs/content/users/install/shell-completion.md

Most people like to have shell completion on the command line. In other words, when you're typing a command, you can hit `<TAB>` and the shell will show you what the options are. For example, if you type `ddev <TAB>`, you'll see all the possible commands. `ddev debug <TAB>` will show you the options for the command. And `ddev list -<TAB>` will show you all the flags available for [`ddev list`](../usage/commands.md#list).

Shells like Bash and Zsh need help to do this though, they have to know what the options are. DDEV provides the necessary hint scripts, and if you use Homebrew, they get installed automatically.

## macOS Bash + Homebrew

> The easiest way to use Bash completion on macOS is install it with Homebrew. `brew install bash-completion`. When you install it though, it will warn you with something like this, which **may vary on your system**. Add the following line to your `~/.bash_profile` file:

Reviewer: ```suggestion
The easiest way to use Bash completion on macOS is install it with Homebrew ([docs](https://docs.brew.sh/Shell-Completion#configuring-completions-in-bash)). `brew install bash-completion`. When you install it though, it will warn you with something like this, which **may vary on your system**. Add the following line to your `~/.bash_profile` file (or if that doesn't exist, to your `~/.profile`:
```

```
[[ -r "$(brew --prefix)/etc/profile.d/bash_completion.sh" ]] && . "$(brew --prefix)/etc/profile.d/bash_completion.sh"
```

!!!note "Bash profile"

>     You must add the include to your `.bash_profile` or `.profile` or nothing will work. Use `source ~/.bash_profile` or `source ~/.profile` to make it take effect immediately.

Reviewer: ```suggestion
    You must add the include to your `.bash_profile` or `.profile` or nothing will work. Use `source ~/.bash_profile` or `source ~/.profile` to make it take effect immediately in your current terminal window.
```

> Link completions by running `brew completions link`.

Reviewer: This seems to be a one-time configuration, and people would always want it. This also seems to affect all types of completions. I imagine we should be telling people at the beginning of each section to do this.

> Shell completions for external Homebrew commands are not automatically installed. To opt-in to using completions for external commands (if provided), they need to be linked to HOMEBREW_PREFIX by running brew completions link.

> When you install DDEV via Homebrew, each new release will automatically get a refreshed completions script.

Reviewer: I don't think we need to say this?

```suggestion
```

> ## macOS Zsh + Homebrew

Reviewer: ```suggestion
## macOS Zsh with Homebrew
```

> To make Homebrew completions available in Zsh the Homebrew-managed path `zsh/site-functions` has to be added to the FPATH. There are two scenarios:

Reviewer: ```suggestion
To make Homebrew completions available in Zsh the Homebrew-managed path `zsh/site-functions` has to be added to the FPATH. There are two scenarios, plain zsh and Oh-My-Zsh, see tabs below:
```

Author: I would vote for sticking with my version for two reasons. First i would avoid directional language like "see tabs below". That is something that should be avoided if possible. And second I tried to avoid "plain" for the title plus the naming for both is different. It should be Zsh and Oh My Zsh according to each of the upstream repos.

Author: On second thought i would even strike `There are two scenarios:`

=== "macOS Zsh"

>     Add the following block to your `~/.zshrc` file:

Reviewer: ```suggestion
    Add the following block to your `~/.zshrc` file (see [docs](https://docs.brew.sh/Shell-Completion#configuring-completions-in-zsh)):
```

    ```bash
    if type brew &>/dev/null
    then
      FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"
      autoload -Uz compinit
      compinit
    fi
    ```

>     Note that the sourcing of the FPATH has to happen before the Zsh completion index is initialized with `compinit`.

Reviewer: ```suggestion
    Note that the updating of the FPATH variable has to happen before the Zsh completion index is initialized with `compinit`.
```

Author: at first i thought sourcing would be the more appropriate and clear term but i guess updating is fine as well and in particular easier to understand.

=== "macOS Zsh with Oh My Zsh"

>     Oh-My-Zsh is calling `compinit` for you when `oh-my-zsh.sh` is sourced. Instead of adding the block that was necessary for `Plain macOS Zsh` place the following line right before `oh-my-zsh.sh` is sourced:

Reviewer: ```suggestion
    Oh-My-Zsh is calling `compinit` for you when `oh-my-zsh.sh` is sourced. Instead of adding the block that was necessary for  simple `macOS Zsh` place the following line right before `oh-my-zsh.sh` is sourced:
```

    ```bash
    FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"
    ```

> To avoid any potential caching issue forcibly rebuild the `.zcompdump` file, which is the index for Zsh completions:

Reviewer: ```suggestion
To avoid any potential caching issue remove and rebuild the `.zcompdump` file, which is the index for Zsh completions:
```

```bash
rm -f ~/.zcompdump; compinit
```

> In case you run into any `zsh compinit: insecure directories` warnings, simply run:

Reviewer: ```suggestion
In case you run into any `zsh compinit: insecure directories` warnings, run:
```

```bash

> chmod -R go-w "$(brew --prefix)/share"

Reviewer: I actually think we should just tell them to do this in the first place.

Author: I will apply that in a follow up that i draft locally and commit after committing your suggestions here

Author: hm on second thought. i think the current order is correct. it is a good thing to reset the index and in case the user runs in the permission issue that would be the next step. so probably the correct order.

```

> Otherwise, you can extract the Zsh completions file (`ddev_zsh_completion.sh`) from the tar archive of completion scripts included with each release. Read more in the [tar archive of completion scripts for manual deployment](#tar-archive-of-completion-scripts-for-manual-deployment) section.

Reviewer: I think this paragraph belongs elsewhere, wherever we describe all the many other possibililties, rather than here.

Author: technically it has nothing to do with macos zsh with homebrew. i've already striked a part but kept that part. but better do a clear separation of concern here.

## Bash/Zsh/Fish on Linux

On Debian and Yum based systems, if you installed DDEV using `apt install ddev`, the `bash`, `zsh`, and `fish` completions should be automatically installed at `/usr/share/bash-completion/completions/ddev`, `/usr/share/zsh/vendor-completions/_ddev` and `/usr/share/fish/completions/ddev.fish` respectively, and the `bash` completions should be automatically installed at `/usr/share/bash-completion/completions/bash`.

Otherwise, you can download the completion files for manual installation as described [below](#tar-archive-of-completion-scripts-for-manual-deployment). Every Linux distro requires a different manual installation technique. On Debian/Ubuntu, you could deploy the `ddev_bash_completion.sh` script where it needs to be by running `sudo mkdir -p /usr/share/bash-completion/completions && sudo cp ddev_bash_completion.sh /usr/share/bash-completion/completions/ddev`.

> ## Fish

Reviewer: Isn't this redundant after we just had Bash/Zsh/Fish on Linux above?

Author: as mentioned on discord the initial pr was intended to only focus on the zsh part and do the other shells in a follow up. and i agree also about that. there has some more untwining happen.

`fish` shell completions are automatically installed at `/usr/local/share/fish/vendor_completions.d/ddev_fish_completion.sh` when you install DDEV via Homebrew.

> If you installed `fish` without Homebrew, you can extract the fish completions (`ddev_fish_completion.sh`) from the tar archive of completion scripts included with each release. See [below](#tar-archive-of-completion-scripts-for-manual-deployment).

Reviewer: If this section is about homebrew and fish, and if it's about macOS, let's label it as such.
